code/modules/projectiles/projectile/beams.dm

```diff
@@ -11,6 +11,16 @@
 	eyeblur = 2
 	impact_effect_type = /obj/effect/temp_visual/impact_effect/red_laser
 	light_color = LIGHT_COLOR_RED
+	ricochets_max = 50	//Honk!
+	ricochet_chance = 80
+
+/obj/item/projectile/beam/check_ricochet(atom/A)
+	. = ..()
+	. = (. | (HAS_SECONDARY_FLAG(A, PROJECTILE_REFLECT)))
+	if(. && prob(ricochet_chance))

```

Comment:
Ok, doesn't stop the fact ```..()``` already does the prob check, which means all of these are far rarer (as they have to pass twice)